tasks/hostname.yml

```diff
@@ -14,10 +14,7 @@
 - name: Set macOS HostName, even when it's an empty string.
   command: "scutil --set HostName '{{ mh_hostname }}'"
   become: true
-  when: >
-    mh_hostname is defined
-    or (
-      mh_current_hostname.stdout is defined
-      and mh_current_hostname.stdout != mh_hostname
-    )
+  when:
+    - "mh_current_hostname.stdout is defined"
```

Comment:
@weakcamel: when the HostName has never been set, what _is_ the stdout of `scutil --get HostName`? (I'm wondering about the `mh_current_hostname.stdout` we've got in there).

Reply:
I'm not sure. I don't have access to a MacOS machine which wasn't already tampered with, I'm afraid.

From Ansible point of view though, stdout being undefined is IMO only possible if the command wasn't executed at all (like, task skipped or command not present)?

Reply:
That said, if this is a good enough simulation then it returns an empty string (+ stderr message) and sets exit code 1:

```shell
$ scutil --set HostName ''
$ scutil --get HostName
HostName: not set
$ echo $?
1
$ scutil --get HostName 2>/dev/null
$
```

Reply:
@ctorgalson  does it look OK to you then? :-)

Reply:
@ctorgalson ping...